File: modules/desktop-client-sdk/src/main/java/com/avandocmsg/messenger/desktop/sdk/call/CallAudioMedia.java
package com.avandocmsg.messenger.desktop.sdk.call;

import java.util.function.Consumer;

public interface CallAudioMedia extends AutoCloseable {

    String createOffer();

    void connect(String answerSdp) throws Exception;

    void sendPcmu(byte[] payload);

    void onPcmu(Consumer<byte[]> listener);

    boolean mediaReady();

    @Override
    void close();
}

File: mobile/mobile-client-sdk/src/commonMain/kotlin/com/avandocmsg/messenger/mobile/sdk/call/CallAudioMedia.kt
package com.avandocmsg.messenger.mobile.sdk.call

interface CallAudioMedia {
    fun createOffer(): String
    suspend fun connect(answerSdp: String)
    fun sendPcmu(payload: ByteArray)
    fun onPcmu(listener: (ByteArray) -> Unit)
    fun mediaReady(): Boolean
    fun close()
}
